app/knights/battle_preparation.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+
+class Knight:
+    instances = []
+
+    def __init__(self, name: str,
+                 power: int,
+                 hp: int,
+                 armour: list,
+                 weapon: dict,
+                 potion: dict = None,
+                 protection: int = 0
```

Comment:
The `protection` attribute should be calculated from the `armour` list rather than being set directly. Consider implementing a method to calculate total protection from the armour parts.